File: sentimental-credit/credit.py
# TODO

def credit():
    is_card = False
    card = input("Number: ")
    card = list(card)
    multi = 0
    for i in range(len(card) - 2, -1, -2):
        top = int(card[i]) * 2
        if top >= 10:
            top1 = top % 10
            top2 = round((top - 4) / 10)
            multi += top1 + top2
            # print(f"{top1}")
            # print(f"{top2}")
        else:
            multi += top
        #     print(f"{top}")
        # print(f"{multi}")
    for i in range(len(card) - 1, -1, -2):
        multi += int(card[i])
        # print(f"{card[i]}")
        # print(f"{multi}")

    # print(f"{multi}")
    if multi % 10 == 0:
        is_card = True
    else:
        print("INVALID")

    if is_card:
        if len(card) == 15 and card[0] == "3":
            if card[1] == "4" or card[1] == "7":
                print("AMEX")
            else:
                print("INVALID")
        elif len(card) == 16 and card[0] == "5":
            if card[1] < "6" and card[1] > "0":
                print("MASTERCARD")
            else:
                print("INVALID")
        elif len(card) == 13 or len(card) == 16:
            if card[0] == "4":
                print("VISA")
            else:
                print("INVALID")
        else:
            print("INVALID")


credit()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Taking Card Number
    long cardnumber;
    do
    {
        cardnumber = get_long("Type your card number: \n");
    }
    while (cardnumber < 0);
    int multi[10], nonmulti[10];
    int j = 0;
    long mastertype = cardnumber;
    long visatype = cardnumber;
    long amextype = cardnumber;
    // Creating Array for calculation
    while (j < 10)
    {
        nonmulti[j] = cardnumber % 10;
        cardnumber = cardnumber / 10;
        multi[j] = cardnumber % 10;
        cardnumber = cardnumber / 10;
        j++;
    }
    int sum1, sum2, sum, sum1m, sum11, sum12;
    for (int sum1i = 0; sum1i < 10; sum1i++)
    {
        sum1m = multi[sum1i] * 2;
        sum11 = sum1m % 10;
        sum12 = (sum1m / 10) % 10;
        sum1 = sum1 + sum11 + sum12;
    }
    for (int sum2i = 0; sum2i < 10; sum2i++)
    {
        sum2 = sum2 + nonmulti[sum2i];
    }
    sum = sum1 + sum2;
    // Controling Card Type
    visatype = visatype / 100000000000;
    while (visatype >= 10)
    {
        visatype /= 10;
    }
    while (mastertype >= 100)
    {
        mastertype /=  100;
    }
    amextype = amextype / 10000000000000;
    // Answer to User
    if (sum % 10 == 0)
    {
        if (amextype == 34 || amextype == 37)
        {
            printf("AMEX\n");
        }
        else if (50 < mastertype && mastertype < 56)
        {
            printf("MASTERCARD\n");
        }
        else if (visatype == 4)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}